service/storage_service.cc

```diff
@@ -3536,7 +3536,7 @@ storage_service::construct_range_to_endpoint_map(
     std::unordered_map<dht::token_range, inet_address_vector_replica_set> res;
     for (auto r : ranges) {
         res[r] = _db.local().find_keyspace(keyspace).get_replication_strategy().get_natural_endpoints(
-                r.end() ? r.end()->value() : dht::maximum_token());
+                r.end() ? r.end()->value() : dht::smallest_token());
```

Comment:
greatest?

Reply:
Same answer as above. Changing `maximum_token` to `smallest_token` as the argument to `calculate_natural_endpoints` is a common pattern in this series.